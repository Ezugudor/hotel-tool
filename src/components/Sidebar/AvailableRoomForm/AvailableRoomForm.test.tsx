import React from 'react';
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import { AvailableRoomForm } from '.';
import { fireEvent } from '@testing-library/react';



let availableRoomFormCont: any;
let onSubmitHandler: Function;
let economyField: any;
let premiumField: any;
let submitButton: any;

beforeAll(() => {
    onSubmitHandler = jest.fn();
    availableRoomFormCont = mount(<AvailableRoomForm onSubmit={onSubmitHandler} />);
    economyField = availableRoomFormCont.find('#economyRoomField').hostNodes();
    premiumField = availableRoomFormCont.find('#premiumRoomField').hostNodes();
    submitButton = availableRoomFormCont.find('button[type="submit"]').hostNodes();
});


describe("AvailableRoomForm Component Test", () => {
    it('should match with existing snapshot', () => {
        expect(availableRoomFormCont.html()).toMatchSnapshot();
    })

    it('should have an input field with id "economyRoomField"', () => {
        expect(economyField).toHaveLength(1);
    })

    it('should have an input field with id "premiumRoomField"', () => {
        expect(premiumField).toHaveLength(1);
    })

    it('should have a button of type "submit"', () => {
        expect(submitButton).toHaveLength(1);
    })

    it('"economyRoomField" should reject non-digit input', () => {
        const input = "-!@#$%^&*().,;'[]{}\\|'\"`~:abcdefghijklmnopqrstuvwxyz";
        economyField.simulate("change", { target: { value: input } });
        expect((economyField.instance()).value).toEqual("");
    });

    it('"premiumRoomField" should reject non-digit input', () => {
        const input = "-!@#$%^&*().,;'[]{}\\|'\"`~:abcdefghijklmnopqrstuvwxyz";
        premiumField.simulate("change", { target: { value: input } });
        expect((premiumField.instance()).value).toEqual("");
    });

    it('should call submit form handler on form submit', () => {
        availableRoomFormCont.simulate('submit', { preventDefault: () => { } });
        expect(onSubmitHandler).toHaveBeenCalledTimes(1)
    })
})
