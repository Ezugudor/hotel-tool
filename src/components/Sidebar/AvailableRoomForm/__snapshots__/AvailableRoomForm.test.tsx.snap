// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailableRoomForm Component Test should match with existing snapshot 1`] = `"<form><div><div><label for=\\"economyRoomField\\">Economy :</label><input type=\\"text\\" name=\\"economy\\" id=\\"economyRoomField\\" value=\\"\\"></div><div><label for=\\"premiumRoomField\\">Premium :</label><input type=\\"text\\" name=\\"economy\\" id=\\"premiumRoomField\\" value=\\"\\"></div><button type=\\"submit\\">Enter</button></div></form>"`;
